src/main/java/com/sda/games/checkers/model/piece/Piece.java

```diff
@@ -21,4 +24,94 @@ public Piece(boolean white) {
+                    System.out.println("No move available!");
+                }
+            } else if (startX == 0 && board.hasPiece(startX + 1, startY + 1)) {
+                if (hasKill(board, player, startX, startY)) {
+                    return true;
+                } else {
+                    System.out.println("No move available!");
+                }
+            } else if (startX == 7 && (board.hasPiece(startX - 1, startY + 1))) {
+                if (hasKill(board, player, startX, startY)) {
+                    return true;
+                } else {
+                    System.out.println("No move available!");
+                }
+            } else if (startX == 7 && (board.hasNoPiece(startX - 1, startY + 1))) {
+                return true;
+            } else if (
+                    board.isNotEmpty(startX + 1, startY + 1) &&
+                            board.isNotEmpty(startX - 1, startY + 1)) {
+                if (hasKill(board, player, startX, startY)) {
+                    return true;
+                } else {
+                    System.out.println("No move available!");
+                }
+            } else {
+                return true;
+            }
+        } else if (!player.isWhite()) {
+            if (board.pieceIsWhite(startX, startY)) {
+                System.out.println("Not your piece!");
+                return false;
+            } else if (startY == 0) {
+                if (hasKill(board, player, startX, startY)) {
+                    return true;
+                } else {
+                    System.out.println("No move available!");
+                }
+            } else if (startX == 0 && (board.isNotEmpty(startX + 1, startY - 1))) {
+                if (hasKill(board, player, startX, startY)) {
+                    return true;
+                } else {
+                    System.out.println("No move available!");
+                }
+            } else if (startX == 7 && (board.isNotEmpty(startX - 1, startY - 1))) {
+                if (hasKill(board, player, startX, startY)) {
+                    return true;
+                } else {
+                    System.out.println("No move available");
+                }
+            } else if (startX == 7 && (board.isEmpty(startX - 1, startY - 1))) {
+                return true;
+            } else if ((startX > 0 && startX < 7) &&
+                    board.isNotEmpty(startX + 1, startY - 1) &&
+                    board.isNotEmpty(startX - 1, startY - 1)) {
+                if (hasKill(board, player, startX, startY)) {
+                    return true;
+                }
+                System.out.println("No move available!");
+            } else {
+                return true;
+            }
+        }
+        return false;
+    }
+
+    public static boolean hasKill(Board board, Player player, int startX, int startY) throws Exception {
+        if (Move.validateExceptionsKilling(board, player, startX, startY)) {
```

Comment:
przeciez to `Move.validateLeftSideKilling` zwraca booleana, mozna to napisac bez ifow, z samymi endami i orami (chyba ze uwazacie bedzie to dla was mniej czytelne wtedy zostawcie tak jak jest :) )